### PR TITLE
Cmake modernization

### PR DESCRIPTION
**Pull request by seanharmer**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -15,25 +15,69 @@ add_subdirectory(source)
...
+install(FILES
+    ${CMAKE_CURRENT_BINARY_DIR}/lunasvgConfig.cmake
+    ${CMAKE_CURRENT_BINARY_DIR}/lunasvgConfigVersion.cmake
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/lunasvg
 )
+
+export(EXPORT lunasvg-targets FILE ${CMAKE_CURRENT_BINARY_DIR}/lunasvgTargets.cmake NAMESPACE lunasvg::)
+
+# Register package in user's package registry
+export(PACKAGE lunasvg)
```
Comment (krf) [inline CMakeLists.txt]: I think we don't usually recommend to do this in FOSS projects, I don't see it as a very popular feature in CMake. Sometimes it comes as a surprise that `find_package` actually searches the user's package registry.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.3)
+cmake_minimum_required(VERSION 3.20)
```
Comment (krf) [inline CMakeLists.txt]: Why such a recent CMake version? This is just a one year old CMake release, won't be available on lots of distro versions.

Side note: Would always add a comment why a certain CMake version was set as minimum required (i.e. `# b/c of feature XZY`)

**Review by krf — COMMENTED** [review]
Rest LGTM

File: CMakeLists.txt [hunk 3]
```diff
@@ -15,25 +15,69 @@ add_subdirectory(source)
...
+include(GNUInstallDirs)
+install(
+    FILES
+    ${CMAKE_CURRENT_SOURCE_DIR}/include/lunasvg.h
+    ${CMAKE_CURRENT_BINARY_DIR}/include/lunasvg_export.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/lunasvg
+)
+install(
+    TARGETS lunasvg
+    EXPORT lunasvg-targets
+    LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
```
Comment (nicolasfella) [inline CMakeLists.txt:52]: IIRC specifying those dirs is redundant/not needed with cmake >=3.14

File: CMakeLists.txt [hunk 4]
```diff
@@ -15,25 +15,69 @@ add_subdirectory(source)
...
+    LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib"
+    RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin"
+    CXX_VISIBILITY_PRESET hidden
+    VISIBILITY_INLINES_HIDDEN 1
+)
+
+add_library(lunasvg::lunasvg ALIAS lunasvg)
+
+if(NOT BUILD_SHARED_LIBS)
+    set_target_properties(lunasvg
```
Comment (nicolasfella) [inline CMakeLists.txt]: Usually I see target_compile_definitions() for this, not sure whether it makes a difference

Comment (nicolasfella) [reply CMakeLists.txt]: CMake recommends using definitions here: https://cmake.org/cmake/help/latest/command/target_compile_options.html#see-also

File: CMakeLists.txt (hunk 1 above)
Comment (seanharmer) [reply CMakeLists.txt]: Thanks, will go with 3.13 as discussed.

File: CMakeLists.txt (hunk 3 above)
Comment (seanharmer) [reply CMakeLists.txt:52]: Will keep them then, since we're going with 3.13 as the baseline.

File: CMakeLists.txt (hunk 1 above)
Comment (seanharmer) [reply CMakeLists.txt]: OK will remove.

File: CMakeLists.txt (hunk 2 above)
Comment (seanharmer) [reply CMakeLists.txt]: Thanks, will go with 3.13 as discussed.

File: CMakeLists.txt (hunk 4 above)
Comment (seanharmer) [reply CMakeLists.txt]: Done, thanks.

**Comment by krf** [comment]
:+1:

**Pull request merged**